app/Http/Controllers/ClassController.php

```diff
@@ -20,23 +20,42 @@ public function index()
         return view('classes', compact ('classes'));    
     }
 
-    /**
+
+
+    /*    #2)
      * Show the form for creating a new resource.
      */
     public function create()
     {
         return view('addclass');
     }
 
-    /**
+    
+
+     /*    #3)
      * Store a newly created resource in storage.
      */
     public function store(Request $request)
     {
     
         #dd($request);
+
+        #validation:
+$data = $request->validate([
+    'classTitle' => "required|string",
+    'capacity' => "required|digits:1",
+    'price' => "required|decimal:1",
+    'begTime' => "required",
```

Comment:
about begTime you can validate `date_format:H:i` remember what users see is not always we store in our DB

i will explain why
from user point of view, he sees that he enters time hours:minutes AM/PM 
from developer point of view, we know that the time enters as 24h format which means we enter hours:minutes

example 
user enters: 8:30 PM
we deal with it as 20:30

Reply:
Ooops,,

I have forget another rule here, what about checking `begTime` to be before `endTime`

there is a rule called `after` & `before`